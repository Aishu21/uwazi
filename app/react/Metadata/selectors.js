import {createSelector} from 'reselect';

import formater from './helpers/formater';

const formatMetadata = createSelector(
  s => s.templates,
  s => s.thesauris,
  (s, d) => d,
  (templates, thesauris, doc) => {
    return formater.prepareMetadata(doc, templates, thesauris).metadata;
  }
);

export { formatMetadata };
