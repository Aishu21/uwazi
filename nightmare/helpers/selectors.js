/*eslint max-len: ["error", 500]*/
export default {
  settingsView: {
    liElementsOfSection: '#app > div.content > div > div > div.col-xs-12.col-sm-8 > div > ul > li',
    firstEditButton: '#app > div.content > div > div > div.col-xs-12.col-sm-8 > div > ul > li:nth-child(1) > div > a > i',
    collectionNameForm: '#app > div.content > div > div > div.col-xs-12.col-sm-8 > div > div.panel-body > form > div > input',
    settingsNavButton: '#app > div.content > header > div > div > ul > li:nth-child(3) > a',
    settingsHeader: '#app > div.content > div > div > div.col-xs-12.col-sm-4 > div > div:nth-child(1) > div.panel-heading',
    logoutButton: '#app > div.content > div > div > div.col-xs-12.col-sm-8 > div > div:nth-child(3) > div.panel-body > a',
    accountButton: '#app > div.content > div > div > div.col-xs-12.col-sm-4 > div > div:nth-child(1) > div.list-group > a:nth-child(1)',
    collectionButton: '#app > div.content > div > div > div.col-xs-12.col-sm-4 > div > div:nth-child(1) > div.list-group > a:nth-child(2)',
    dictionariesButton: '#app > div.content > div > div > div.col-xs-12.col-sm-4 > div > div:nth-child(2) > div.list-group > a:nth-child(3)',
    documentsButton: '#app > div.content > div > div > div.col-xs-12.col-sm-4 > div > div:nth-child(2) > div.list-group > a:nth-child(1)',
    entitiesButton: '#app > div.content > div > div > div.col-xs-12.col-sm-4 > div > div:nth-child(2) > div.list-group > a:nth-child(4)',
    connectionsButton: '#app > div.content > div > div > div.col-xs-12.col-sm-4 > div > div:nth-child(2) > div.list-group > a:nth-child(2)',
    dictionariesBackButton: '#app > div.content > div > div > div.col-xs-12.col-sm-8 > div > form > div > div.panel-heading > a',
    documentsBackButton: '#app > div.content > div > div > div.col-xs-12.col-sm-8 > div > main > div > form > div > a',
    connectionsBackButton: '#app > div.content > div > div > div.col-xs-12.col-sm-8 > div > form > div > div.panel-heading.relationType > a',
    entitiesBackButton: '#app > div.content > div > div > div.col-xs-12.col-sm-8 > div > main > div > form > div > a',
    addNewDictionary: '#app > div.content > div > div > div.col-xs-12.col-sm-8 > div > div.panel-body > a',
    addNewDocument: '#app > div.content > div > div > div.col-xs-12.col-sm-8 > div > div.panel-body > a',
    addNewEntity: '#app > div.content > div > div > div.col-xs-12.col-sm-8 > div > div.panel-body > a',
    addNewConnection: '#app > div.content > div > div > div.col-xs-12.col-sm-8 > div > div.panel-body > a > span',
    addNewValueToDictionaryButton: '#app > div.content > div > div > div.col-xs-12.col-sm-8 > div > form > div > div.panel-body > a',
    firstDictionaryValForm: '#app > div.content > div > div > div.col-xs-12.col-sm-8 > div > form > div > ul > li:nth-child(2) > div > div > input',
    secondDictionaryValForm: '#app > div.content > div > div > div.col-xs-12.col-sm-8 > div > form > div > ul > li:nth-child(3) > div > div > input',
    saveDictionaryButton: '#app > div.content > div > div > div.col-xs-12.col-sm-8 > div > form > div > div.panel-heading > button',
    saveDocumentButton: '#app > div.content > div > div > div.col-xs-12.col-sm-8 > div > main > div > form > div > button',
    saveEntityButton: '#app > div.content > div > div > div.col-xs-12.col-sm-8 > div > main > div > form > div > button',
    saveConnectionButton: '#app > div.content > div > div > div.col-xs-12.col-sm-8 > div > form > div > div.panel-heading.relationType > button',
    dictionaryNameForm: '#thesauriName',
    connectionNameForm: '#relationTypeName',
    entityNameForm: '#app > div.content > div > div > div.col-xs-12.col-sm-8 > div > main > div > form > div > div > input',
    documentTemplateNameForm: '#app > div.content > div > div > div.col-xs-12.col-sm-8 > div > main > div > form > div > div > input',
    deleteButtonConfirmation: 'body > div.ReactModalPortal > div > div > div > div.modal-footer > button.btn.confirm-button.btn-danger'
  },
  libraryView: {
    unlinkIcon: '#app > div.content > div > div > aside.side-panel.document-metadata.is-active > div.sidepanel-body > div > div.tab-content.tab-content-visible > div > div.item.relationship-active > div.item-actions > a:nth-child(1)',
    libraryFirstDocument: '.item-group .item',
    searchInLibrary: '#app > div.content > header > div > div > div > a > i.fa.fa-search',
    firstDocumentViewButton: '#app > div.content > div > div > main > div > div.item-group > div:nth-child(1) > div.item-actions > a',
    libraryNavButton: '#app > div.content > header > div > div > ul > li:nth-child(1) > a',
    loadTargetDocumentButton: '#app > div.content > div > div > aside.side-panel.create-reference.is-active > div.sidepanel-footer > button'
  },
  uploadsView: {
    uploadsNavButton: '.fa-cloud-upload',
    uploadBox: '#app > div.content > div > div > main > div:nth-child(1) > div',
    uploadsBottomRightSaveButton: '.float-btn'
  }
};
