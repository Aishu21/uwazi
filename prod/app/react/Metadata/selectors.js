"use strict";Object.defineProperty(exports, "__esModule", { value: true });exports.metadataSelectors = void 0;var _reselect = require("reselect");

var _immutable = _interopRequireDefault(require("immutable"));
var _formater = _interopRequireDefault(require("./helpers/formater"));function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

const indexValues = (t) =>
t.set('values', t.get('values').reduce((indexed, value) => {
  if (value.get('values')) {
    return indexed.merge(indexValues(value).get('values'));
  }
  return indexed.set(value.get('id'), value);
}, new _immutable.default.Map({})));

const indexedThesaurus = (0, _reselect.createSelector)(
s => s.thesauris,
thesaurus => thesaurus.map(t => indexValues(t)));


const formatMetadata = (0, _reselect.createSelector)(
s => s.templates,
indexedThesaurus,
(s, doc, sortProperty, references) => ({ doc, sortProperty, references }),
(templates, thesauris, { doc, sortProperty, references }) => {
  if (sortProperty) {
    return _formater.default.prepareMetadataForCard(doc, templates, thesauris, sortProperty).metadata;
  }
  return _formater.default.prepareMetadata(doc, templates, thesauris, references).metadata;
});


const metadataSelectors = {
  formatMetadata,
  indexedThesaurus };exports.metadataSelectors = metadataSelectors;