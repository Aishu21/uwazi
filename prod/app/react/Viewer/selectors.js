"use strict";Object.defineProperty(exports, "__esModule", { value: true });exports.parseReferences = exports.selectTargetReferences = exports.selectTargetDoc = exports.selectReferences = exports.selectDoc = void 0;var _reselect = require("reselect");
var _immutable = require("immutable");

const documentViewer = s => s.documentViewer;

function isRelationshipAReference(doc, reference) {
  return reference.get('entity') === doc.get('sharedId') && typeof reference.getIn(['range', 'start']) !== 'undefined';
}

const parseReferences = (doc, refs) => refs.
filter(r => isRelationshipAReference(doc, r)).
reduce((hubs, r) => {
  if (hubs.indexOf(r.get('hub')) === -1) {
    hubs.push(r.get('hub'));
  }
  return hubs;
}, []).
reduce((memo, hubId) => {
  let references = memo;
  const baseReference = refs.find(r => isRelationshipAReference(doc, r) && r.get('hub') === hubId);
  const otherRelationshipsOfHub = refs.filter(r => r.get('_id') !== baseReference.get('_id') && r.get('hub') === hubId);
  otherRelationshipsOfHub.forEach(r => {
    references = references.push(baseReference.set('associatedRelationship', r));
  });
  return references;
}, (0, _immutable.fromJS)([]));exports.parseReferences = parseReferences;

const selectDoc = (0, _reselect.createSelector)(s => documentViewer(s).doc, doc => doc);exports.selectDoc = selectDoc;
const selectRefs = (0, _reselect.createSelector)(s => documentViewer(s).references, references => references);

const selectReferences = (0, _reselect.createSelector)(
selectDoc,
selectRefs,
parseReferences);exports.selectReferences = selectReferences;


const selectTargetDoc = (0, _reselect.createSelector)(s => documentViewer(s).targetDoc, doc => doc);exports.selectTargetDoc = selectTargetDoc;
const selectTargetRefs = (0, _reselect.createSelector)(s => documentViewer(s).targetDocReferences, references => references);

const selectTargetReferences = (0, _reselect.createSelector)(
selectTargetDoc,
selectTargetRefs,
parseReferences);exports.selectTargetReferences = selectTargetReferences;