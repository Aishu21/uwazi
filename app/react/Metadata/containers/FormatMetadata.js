import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import React from 'react';

import { formatMetadata } from '../selectors';
import Metadata from '../components/Metadata';


const FormatMetadata = props => (
  <Metadata metadata={formatMetadata(props, props.entity, props.sortedProperty)} compact={!!props.sortedProperty}/>
);

FormatMetadata.defaultProps = {
  sortedProperty: ''
};

FormatMetadata.propTypes = {
  entity: PropTypes.object.isRequired,
  sortedProperty: PropTypes.string
};

export function mapStateToProps(state, { entity, sortedProperty }) {
  return {
    templates: state.templates,
    thesauris: state.thesauris,
    entity,
    sortedProperty
  };
}

export default connect(mapStateToProps)(FormatMetadata);
