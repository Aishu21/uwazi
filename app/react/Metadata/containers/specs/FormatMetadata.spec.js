import React from 'react';

import { shallow } from 'enzyme';

import FormatMetadata from '../FormatMetadata';
import * as selectors from '../../selectors';

describe('FormatMetadata', () => {
  it('should render Metadata component passing the formatted metadata', () => {
    spyOn(selectors, 'formatMetadata').and.returnValue(['formated', 'metadata']);
    const props = {
      templates: [],
      thesauris: [],
      entity: {},
      sortedProperty: 'sortedProperty'
    };
    const component = shallow(<FormatMetadata.WrappedComponent {...props} />);
    expect(component).toMatchSnapshot();
  });
});
