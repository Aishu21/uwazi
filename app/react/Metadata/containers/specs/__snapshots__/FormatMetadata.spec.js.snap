// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormatMetadata should render Metadata component passing the formatted metadata 1`] = `
<Metadata
  compact={true}
  metadata={
    Array [
      "formated",
      "metadata",
    ]
  }
/>
`;

exports[`FormatMetadata should unshift additional metadata if passed 1`] = `
<Metadata
  compact={true}
  metadata={
    Array [
      "more",
      "data",
      "formated",
      "metadata",
    ]
  }
/>
`;
